src/msac.rs

```diff
@@ -301,38 +296,40 @@ unsafe fn dav1d_msac_decode_symbol_adapt_rust(
 
 unsafe extern "C" fn dav1d_msac_decode_symbol_adapt_c(
     s: *mut MsacContext,
-    cdf: *mut uint16_t,
+    cdf: *mut u16,
     n_symbols: size_t,
 ) -> libc::c_uint {
-    dav1d_msac_decode_symbol_adapt_rust(s, cdf, n_symbols)
+    // # Safety
+    //
+    // This is only called from [`dav1d_msac_decode_symbol_adapt16`],
+    // where there is an `assert!(n_symbols < cdf.len());`.
+    // Thus, `n_symbols + 1` is a valid length for the slice `cdf` came from.
+    #[deny(unsafe_op_in_unsafe_fn)]
```

Comment:
Why is this a `#[deny]` attribute? Shouldn't this be an `#[allow]`?

Reply:
Without this, `rustc` says there's an unused `unsafe` in this function since the function itself is `unsafe`.  But the function itself has to stay `unsafe` for the `symbol_adapt16` `fn` ptr.  I want to be clear this is proven safe with an `unsafe` block and `Safety` comment.  `#[deny(unsafe_op_in_unsafe_fn)]` tells `rustc` that `unsafe` operations are not allowed just because a function is `unsafe` and that `unsafe` blocks are still needed.